src/Rules.Framework.Providers.MongoDb/MongoDbRulesDataSourceInitializer.cs

```diff
@@ -0,0 +1,73 @@
+namespace Rules.Framework.Providers.MongoDb
+{
+    using System;
+    using System.Globalization;
+    using System.Threading.Tasks;
+    using MongoDB.Driver;
+    using Rules.Framework.Providers.MongoDb.DataModel;
+
+    internal static class MongoDbRulesDataSourceInitializer
+    {
+        private static bool isInitialized;
+
+        public static async Task InitializeAsync(IMongoClient mongoClient, MongoDbProviderSettings mongoDbProviderSettings)
+        {
+            if (isInitialized)
+            {
+                return;
+            }
+
+            if (mongoClient is null)
+            {
+                throw new ArgumentNullException(nameof(mongoClient));
+            }
+
+            if (mongoDbProviderSettings is null)
+            {
+                throw new ArgumentNullException(nameof(mongoDbProviderSettings));
+            }
+
+            var mongoDatabase = mongoClient.GetDatabase(mongoDbProviderSettings.DatabaseName, new MongoDatabaseSettings
+            {
+                ReadConcern = ReadConcern.Available,
+                ReadPreference = ReadPreference.SecondaryPreferred,
+                WriteConcern = WriteConcern.Acknowledged,
+            });
+
+            // Creates rules collection if it does not exist.
```

Comment:
Instead of adding a comment, can you add a private method describing its purpose?

Reply:
Agree, done 👍